stl/inc/stdatomic.h

```diff
@@ -0,0 +1,125 @@
+// stdatomic.h standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _STDATOMIC_H_
+#define _STDATOMIC_H_
+#include <yvals.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#ifdef _M_CEE_PURE
+#error <stdatomic.h> is not supported when compiling with /clr:pure.
+#endif // _M_CEE_PURE
+
+#if !_HAS_CXX23
+#pragma message("The contents of <stdatomic.h> are available only with C++23 or later.")
+#else // ^^^ !_HAS_CXX23 / _HAS_CXX23 vvv
+
+#include <atomic>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+template <class _Ty>
+using _Std_atomic = _STD atomic<_Ty>;
+
+#define _Atomic(T) _Std_atomic<T>
+
+// clang-format off
+using _STD memory_order;
+using _STD memory_order_relaxed;
+using _STD memory_order_consume;
+using _STD memory_order_acquire;
+using _STD memory_order_release;
+using _STD memory_order_acq_rel;
+using _STD memory_order_seq_cst;
+
+using _STD atomic_flag;
+

```

Comment:
`atomic_bool` and `atomic_char8_t` are depicted in WG21-N4892 but are missing here. (Similarly in the test.)